docs/components/styling.md

```diff
@@ -233,6 +233,66 @@ By specifying "inner-text" as the value of the `exportparts` attribute, elements
 <outer-component />
 ```
 
+## Style Modes
+
+Component Style Modes enable you to create versatile designs for your components by utilizing different styling configurations. This is achieved by assigning the styleUrls property of a component to a collection of style mode names, each linked to their respective CSS files.
+
+### Example: Styling a Button Component
+
+Consider a basic button component that supports both iOS and Material Design aesthetics:
+
+```tsx
+@Component({
+  tag: 'simple-button',
+  styleUrls: {
+    md: './simple-button.md.css', // styles for Material Design
+    ios: './simple-button.ios.css' // styles for iOS
+  },
+})
+export class SimpleButton {
+  // ...
+}
+```
+
+To dictate the style mode (Material Design or iOS) in which the button should be rendered, you must initialize the desired mode before any component rendering occurs. This can be done as follows:
+
+```ts
+import { setMode } from '@stencil/core';
+setMode(() => 'ios'); // Setting iOS as the default mode for all components
+```
+
+The `setMode` function processes all elements, enabling the assignment of modes individually based on specific element attributes. For instance, by assigning the `colorMode` attribute to a component:
+
+```html
+<simple-button colorMode="ios"></simple-button>
+```
+
+You can conditionally set the style mode based on the `colorMode` property:
+
+```ts
+import { setMode } from '@stencil/core';
+
+const defaultMode = 'md'; // Default to Material Design
+setMode((el) => el.getAttribute('colorMode') || defaultMode);
+```
+
+Style modes can only be set once at the beginning of the component lifecycle.
+
+### Important Considerations
+
+- __Initialization:__ Style modes must be defined at the start of the component lifecycle and cannot be changed thereafter.
```

Comment:
I think we should expand on this a bit and clarify what "beginning of the component lifecycle" means. Like does this need to happen before bootstrap? Can it be set in any lifecycle callbacks?

Reply:
Updated the wording to:

```
- __Initialization:__ Style modes must be defined before the components gets rendered into the DOM and can't be changed afterwards. Updating the style mode requires manually re-rendering of all components.
```

If you update a style mode you have to re-render the whole component, meaning: remove it from the DOM and add it back in. Maybe there is a helper method that could do that, something like `requestUpdate` in Lit? Stencil has to re-initialize the component including its styles.

I may suggest two options here:
- don't recommend to ever update the mode once initially set
- ingest a bug ticket into our roadmap to investigate mode updates during runtime